pkg/services/sqlstore/migrations/ualert/silences.go

```diff
@@ -0,0 +1,107 @@
+package ualert
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/matttproud/golang_protobuf_extensions/pbutil"
+	pb "github.com/prometheus/alertmanager/silence/silencepb"
+
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+)
+
+func (m *migration) addSilence(da dashAlert, rule *alertRule) error {
+	if da.State != "paused" {
+		return nil
+	}
+
+	uid, err := uuid.NewV4()
```

Comment:
is it necessary to introduce this new dependency? otherwise I would suggest using the [util.GenerateShortUID()](https://github.com/grafana/grafana/blob/b7628f2060c58ee9838f40bb26289881e47ef3b5/pkg/util/shortid_generator.go#L24)

Reply:
Alertmanager seems to be using this package for UID, not sure how things could break if we use a different UID here. So I feel it would be safer to go with this one. Also, we already indirectly depend on this hence `go.sum` was not updated and only `go.mod`.